### ISSUE TITLE
[Bug]: localizedField in Blocks return error getClass on object in Localizedfield.php

### ISSUE THREAD

**Issue by rolfhermancoen** [pimcore/pimcore#15529]
### Pimcore version

11.0.3

### Steps to reproduce

 - Create a class with the following attributes:

1. Block
2. localizedField
3. input

![Screenshot 2023-07-13 at 20 20 00](https://github.com/pimcore/pimcore/assets/31653689/c60fc510-e860-4c5f-826c-2a80a8c67e40)

- Load up a data object
- Add data in the blocks
- Load up the data object again
- Get an error

### Expected / actual behavior

**Expected:**
The data object loads normally and the localized fields are visible in the blocks.

**Actual Behavior:**
The loading of the data object returns an error. The object is null when loading the data in Localizedfield.php, where loading the class through getClass also returns null and therefore returns an error.

**Possible fix:**
Currently I fixed it through creating a different method, called getDataObject, where the object is retrieved through the objectId, which is actually present in the context of the Localizedfield. This is then called only when the the containerTyp is block.

```
../models/DataObject/Localizedfield.php
@@ -243,6 +243,11 @@
         return $this->object;
     }
 
+    public function getDataObject(): ?Concrete
+    {
+        return DataObject::getById($this->objectId);
+    }
+
     public function setClass(?ClassDefinition $class): static
     {
         $this->class = $class;
@@ -309,7 +314,7 @@
             $container = Model\DataObject\Objectbrick\Definition::getByKey($containerKey);
         } elseif (isset($context['containerType']) && $context['containerType'] === 'block') {
             $containerKey = $context['containerKey'];
-            $object = $this->getObject();
+            $object = $this->getDataObject();
             $blockDefinition = $object->getClass()->getFieldDefinition($containerKey);
             $container = $blockDefinition;
         } else {
@@ -349,7 +354,8 @@
             $container = $brickDef->getFieldDefinition('localizedfields');
         } elseif (isset($context['containerType']) && $context['containerType'] === 'block') {
             $containerKey = $context['containerKey'];
-            $object = $this->getObject();
+            $object = $this->getDataObject();
             /** @var Model\DataObject\ClassDefinition\Data\Block $block */
             $block = $object->getClass()->getFieldDefinition($containerKey);
             /** @var Model\DataObject\ClassDefinition\Data\Localizedfields $container */

```

**Comment by fashxp** [comment]
Probably related to https://github.com/pimcore/pimcore/issues/14090

We must not load dataobjects with getbyid in that case, as this causes other issues, see also https://github.com/pimcore/pimcore/pull/14089

We need to somehow set the current object from outside..

**Comment by kingjia90** [comment]
Had some doubt on https://github.com/pimcore/pimcore/pull/14291/files?w=1#diff-bcff749d99aba6e16e7b74921e17069cdb2b38c5e11c77f28bb0d88cec7ef44aL244-L247 removal because is not consinstent with this part which is kept on https://github.com/pimcore/pimcore/blob/0cee68048c70ac9696c2a8ba195fb26b596a9467/models/DataObject/Objectbrick.php#L187-L193 ?

Reverting the lines in that PR, things works again but not sure if it the other way round, that in object brick missed the same removal.

**Comment by fashxp** [comment]
The latter should be removed too. 
As said above, we must not load dataobjects with getbyid in that case, as this causes other issues, see also https://github.com/pimcore/pimcore/pull/14089

**Comment by kingjia90** [comment]
https://github.com/pimcore/pimcore/pull/15868 seems could be fixing the issue

**Comment by dvesh3** [comment]
Fixed by #15868
